src/main/java/com/adobe/aem/compgenerator/utils/HTMLUtils.java

```diff
@@ -0,0 +1,103 @@
+package com.adobe.aem.compgenerator.utils;
+
+import com.adobe.aem.compgenerator.Constants;
+import com.adobe.aem.compgenerator.models.GenerationConfig;
+import com.adobe.aem.compgenerator.models.Property;
+import org.apache.commons.lang3.StringUtils;
+
+import java.util.Collection;
+import java.util.List;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+public class HTMLUtils {
+
+    public static String renderHtml(GenerationConfig generationConfig) {
+        List<Property> props = getAllProperties(generationConfig);
+        String slingModel = StringUtils.uncapitalize(generationConfig.getJavaFormatedName()) + "Model";
+        StringBuilder renderedHtml = new StringBuilder();
+
+        for (Property prop : props) {
+            String label = "label_not_defined";
+            if (prop.getLabel() != null) {
+                label = prop.getLabel();
```

Comment:
Looks like this code is having trouble grabbing the label for checkbox and radio button fields:
```
<p>label_not_defined: ${demoCompModel.checkTest ? 'checked' : 'unchecked'}</p>
<p>label_not_defined: ${demoCompModel.radioTest}</p>
<p>label_not_defined: ${demoCompModel.hiddenTest}</p>
```

Also, hidden field cannot find a label as well, but perhaps we should use something like "Hidden field <fieldname>:"